feat: Add BeInRange assertion for numeric range validation

lvchenjia opened the pull request:
## Body:

This pull request introduces the BeInRange assertion, a new feature designed to simplify numeric range validation in tests. It provides a clear and concise way to assert that a value falls within an inclusive [min, max] range.

## Problem

Closes #12 

Previously, validating if a number was within a specific range required multiple assertions (e.g., BeGreaterOrEqualTo and BeLessOrEqualTo). This approach was verbose and produced generic error messages that didn't clearly state that a range check had failed. This is a common requirement for validating things like age (18-65), scores (0-100), or HTTP status codes (200-299).

## Solution

I have introduced a new assertion, BeInRange, that addresses this gap.
New Assertion:
assert.BeInRange[T Ordered] provides the core logic in assert/assertions.go.
should.BeInRange serves as the user-facing wrapper in should.go.

Andrei-hub11 commented on assert/assertions.go:
```diff
@@ -426,6 +426,52 @@ func BeLessOrEqualTo[T Ordered](t testing.TB, actual T, expected T, opts ...Opti
+		messageBuilder.WriteString("\n")
+	}
+
+	if actual < min {
+		messageBuilder.WriteString(fmt.Sprintf("Expected value to be in range [%v, %v], but it was below:", min, max))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Value    : %v", actual))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Range    : [%v, %v]", min, max))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Distance : %v below minimum (%v < %v)", min-actual, actual, min))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Hint     : Value should be >= %v", min))
+	} else { // actual > max
+		messageBuilder.WriteString(fmt.Sprintf("Expected value to be in range [%v, %v], but it was above:", min, max))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Value    : %v", actual))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Range    : [%v, %v]", min, max))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Distance : %v above maximum (%v > %v)", actual-max, actual, max))
+		messageBuilder.WriteString(fmt.Sprintf("\n        Hint     : Value should be <= %v", max))
+	}
```
I think you should extract a specialized function for that, and formatRangeError sounds like a good name. Also, make sure to cover it with some tests. After that, I think everything will be complete.

lvchenjia commented:
Thanks, @Andrei-hub11. I've made the requested changes. Ready for another look.

Andrei-hub11 commented on assert/assetions_test.go:
```diff
@@ -5077,3 +5077,113 @@ func TestFail_Integration(t *testing.T) {
 	})
 }
+
+func TestBeInRange(t *testing.T) {
```
Just one more thing — I noticed that one of the subtests already checks the custom message, but could you add a dedicated subtest specifically to verify that the custom message appears in the error for this assertion?

After that, I’ll approve it. Then we can just wait to see if @ccoVeille  has any comments.

```
t.Run("Custom messages", func(t *testing.T) {
})
```

ccoVeille replied:
I'm sorry I'm unsure what you meant

Andrei-hub11 replied:
I meant you might end up adding some additional review 😅

ccoVeille commented on assert/utils.go:
```diff
@@ -2145,3 +2145,21 @@ func formatMapNotContainValueError(target interface{}, mapValue interface{}) str
 	return msg.String()
 }
+
+func formatRangeError[T Ordered](actual, min, max T) string {
```
Same about minValue, maxValue here


```suggestion
func formatRangeError[T Ordered](actual, minValue, maxValue T) string {
```

ccoVeille commented on assert/utils_test.go:
```diff
@@ -2633,3 +2633,75 @@ func TestRemoveDuplicateSimilarItems(t *testing.T) {
 	})
 }
+
+func TestFormatString(t *testing.T) {
+	// ... existing test ...
+}
```
I'm unsure I get this

ccoVeille commented on assert/assertions.go:
```diff
@@ -426,6 +426,37 @@ func BeLessOrEqualTo[T Ordered](t testing.TB, actual T, expected T, opts ...Opti
 	}
 }
 
+// BeInRange reports a test failure if the value is not within the specified range (inclusive).
```
```suggestion
// BeBetween reports a test failure if the value is not within the specified range (inclusive).
```

What about this? Just a suggestion, I'm unsure what would be the clearest

lvchenjia commented:
Thank you for your feedback. I've pushed a new commit to address the points you raised.
I've made the changes based on my understanding, but I'm not entirely certain if I've captured your intent perfectly. Could you please take another look?

Andrei-hub11 commented:
@lvchenjia  ccoVeille  suggested some adjustments that didn’t come in your latest commit

Andrei-hub11 commented:
@lvchenjia, you forgot to remove this on line 2637 of utils_test.go:

```
func TestFormatString(t *testing.T) {
	// ... existing test ...
}
```

Andrei-hub11 commented:
@lvchenjia Could you update the README with examples for BeInRange? After removing that part from the previous comment and adding the examples, everything will be good on my end.

lvchenjia commented:
My apologies for the oversight.
I've made the requested changes:
Deleted the extra test function in utils_test.go.
Added the BeInRange example to the README.
Ready for another look. Thanks.

Andrei-hub11 commented:
@lvchenjia Alright. Very good. Thanks. Let’s just wait to see if @ccoVeille  has any additional comments. If not, I’ll go ahead and merge it.

Andrei-hub11 commented:
Thanks again. @lvchenjia
